homeassistant/components/canary/alarm_control_panel.py

```diff
@@ -23,11 +23,11 @@
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Canary alarms."""
-    data = hass.data[DATA_CANARY]
-    devices = []
+    if discovery_info is not None:
```

Comment:
Why should we load the platforms via discovery if we don't want that?

Reply:
The camera platform does need it, reverted the check from other platforms.